pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -345,11 +345,15 @@ func (s *Controller) workloadEntryHandler(old, curr config.Config, event model.E
 	// update eds cache only
 	s.edsUpdate(allInstances, false)
 
+	// TODO: if len(configsUpdated) == 0we're firing a Forced push for compatibility because no configs have been updated,
```

Comment:
These seem like we should just do nothing if there are no configs updated

Reply:
I thought so, but don't know if it could cause unwanted effects

Reply:
looking at the handler's code it would seem that it's not that simple, there seem to be a lot of changes happening even when not updating `configsUpdated`, so I don't think it's currently safe to skip pushes if `configsUpdated` si empty, we should probably refactor these handlers.